hail/Makefile

```diff
@@ -0,0 +1,192 @@
+include env_var.mk
+
+MAKEFLAGS += --no-builtin-rules
+.SUFFIXES:
+
+PARALLELISM ?= $(shell python -c 'import multiprocessing; print(multiprocessing.cpu_count())')
+SPARK_VERSION ?= 2.2.0
+# Hail has three notions of version:
+#  - hail short version: MAJOR.MINOR
+#  - hail pip version: MAJOR.MINOR.PATCH
+#  - hail version: MAJOR.MINOR.PATCH-GIT_SHA, calculated by generateBuildInfo
+MAJOR ?= 0
+MINOR ?= 2
+PATCH ?= 8
+HAIL_PIP_VERSION := $(MAJOR).$(MINOR).$(PATCH)$(PIP_VERSION_SUFFIX)
+GRADLE_ARGS := $(GRADLE_ARGS) -Dspark.version=$(SPARK_VERSION)
+GIT_SHA := $(shell git rev-parse HEAD)
+
+$(eval $(call ENV_VAR,SPARK_VERSION))
+$(eval $(call ENV_VAR,HAIL_PIP_VERSION))
+$(eval $(call ENV_VAR,GRADLE_ARGS))
+$(eval $(call ENV_VAR,GIT_SHA))
+
+IGNORED_FILE_PATTERN := \( -name '*~' \)
+JVM_FILES := $(shell find src/main/java src/main/scala src/main/resources \( -not $(IGNORED_FILE_PATTERN) \))
+IGNORED_PY_FILE_PATTERN = \( $(IGNORED_FILE_PATTERN) -o -name '*.pyc' \)
```

Comment:
`hail-*.log`?

Maybe we should add a task to delete log files?

Reply:
i didn't see any files in our hierarchy that end in `.log` so I ignored all `.log` files. Also added this to the .gitignore.

Reply:
I like having a task to delete log files. Wanna PR that after this one lands?

Reply:
sure, sounds good.